java/piranha/src/main/java/com/uber/piranha/config/Config.java

```diff
@@ -124,17 +130,49 @@ private Config(
   }
 
   /**
-   * Return all configuration method records matching a given method name.
+   * Return all configuration method records matching the name of the given method invocation tree.
+   * If the cleanup option "allow_method_chain" is set to true, returns all configurations method
+   * record matching the name of the given method invocation and its receiver method invocation. For
+   * instance, the invocation `exp.stale_flag().getValue()` will match the method record with name
+   * as "stale_flag.getValue". Note: This method only supports matching a method chain of length 2.
    *
-   * @param methodName the method name to search
+   * @param mit Method invocation AST
    * @return A collection of {@link PiranhaMethodRecord} objects, representing each method
    *     definition in the piranha json configuration file matching {@code methodName}.
    */
-  public ImmutableCollection<PiranhaMethodRecord> getMethodRecordsForName(String methodName) {
-    return configMethodProperties.containsKey(methodName)
-        ? configMethodProperties.get(methodName)
-        : ImmutableSet.of();
+  public ImmutableCollection<PiranhaMethodRecord> getMethodRecordsForName(
+      MethodInvocationTree mit) {
+    String methodName = getMethodName(mit);
+    if (configMethodProperties.containsKey(methodName)) {
+      return configMethodProperties.get(methodName);
+    }
+    if (allowMethodChain() && mit.getMethodSelect() instanceof MemberSelectTree) {
+      ExpressionTree mstExpr = ((MemberSelectTree) mit.getMethodSelect()).getExpression();
+      if (mstExpr instanceof MethodInvocationTree) {
+        String chainedMethodName = getMethodName((MethodInvocationTree) mstExpr) + "." + methodName;
+        if (configMethodProperties.containsKey(chainedMethodName)) {
+          return configMethodProperties.get(chainedMethodName);
+        }
+      }
+    }
+    return ImmutableSet.of();
+  }
+
+  /**
+   * Returns name of the method invocation
+   *
+   * @param mit Method invocation tree
+   * @return a string representing the name of the method invocation
+   */
+  public String getMethodName(MethodInvocationTree mit) {
+    if (mit.getMethodSelect() instanceof MemberSelectTree) {
+      MemberSelectTree mst = (MemberSelectTree) mit.getMethodSelect();
+      return mst.getIdentifier().toString();
+    }
+    // scenario when there is no receiver, like foo("bar")
```

Comment:
what will happen in the invocation of this at line 152? Please consider adding a related test for this.

Reply:
Good catch. :)